src/Compilers/CSharp/Test/EndToEnd/EndToEndTests.cs

```diff
@@ -723,5 +725,57 @@ class XAttribute : System.Attribute { }
             Assert.Collection(runResult.TrackedSteps["result_ForAttributeWithMetadataName"],
                 step => Assert.True(step.Outputs.Single().Value is ClassDeclarationSyntax { Identifier.ValueText: "C1" }));
         }
+
+        [Fact]
+        public void ManyBinaryPatterns()
+        {
+            const string preamble = $"""
+                int i = 2;
+
+                System.Console.Write(i is
+                """;
+            const string append = $"""
+
+                or 
```

Comment:
>or [](http://example.com/codeflow?start=16&length=2)

Consider testing deep nesting of `and` patterns as well since there is some difference in `Binder.BindBinaryPattern()`.
```csharp
Console.Write(i is not 1 and not 2 and ...);
```